### ISSUE TITLE
java.lang.String cannot be cast to com.facebook.react.uimanager.AccessibilityDelegateUtil$AccessibilityRole - on 0.57.3

### ISSUE THREAD

**Issue by Ananlabs** [facebook/react-native#21754]
## Environment
<pre><code>
  React Native Environment Info:
    System:
      OS: Windows 10
      CPU: x64 Intel(R) Core(TM) i5-5200U CPU @ 2.20GHz
      Memory: 1.04 GB / 5.92 GB
    Binaries:
      Yarn: 1.10.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 5.6.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.0.0.0 AI-171.4443003
</code></pre>

## Description
<pre><code>
java.lang.String cannot be cast to com.facebook.react.uimanager.AccessibilityDelegateUtil$AccessibilityRole
setDelegate
    AccessibilityDelegateUtil.java:93
updateViewAccessibility
    BaseViewManager.java:260
onAfterUpdateTransaction
    BaseViewManager.java:266
updateProperties
    ViewManager.java:33
createView
    NativeViewHierarchyManager.java:269
execute
    UIViewOperationQueue.java:200
run
    UIViewOperationQueue.java:888
flushPendingBatches
    UIViewOperationQueue.java:1001
access$2400
    UIViewOperationQueue.java:46
doFrameGuarded
    UIViewOperationQueue.java:1061
doFrame
    GuardedFrameCallback.java:29
doFrame
    ReactChoreographer.java:134
doFrame
    ChoreographerCompat.java:105
run
    Choreographer.java:856
doCallbacks
    Choreographer.java:670
doFrame
    Choreographer.java:603
run
    Choreographer.java:844
handleCallback
    Handler.java:739
dispatchMessage
    Handler.java:95
loop
    Looper.java:148
main
    ActivityThread.java:5417
invoke
    Method.java
run
    ZygoteInit.java:726
main
    ZygoteInit.java:616

</code></pre>
## Dependencies
<pre><code>
  "dependencies": {
    "axios": "^0.18.0",
    "native-base": "^2.8.1",
    "react": "16.6.0-alpha.8af6728",
    "react-native": "0.57.3",
    "react-navigation": "^2.18.0",
    "react-redux": "^5.0.7",
    "redux": "^4.0.0",
    "redux-thunk": "^2.3.0"
  }
</code></pre>

**Comment by diggis00** [comment]
Same problem
![whatsapp image 2018-10-12 at 8 41 20 pm](https://user-images.githubusercontent.com/30402460/46877816-577fac00-ce5f-11e8-8e18-e82b89263c5c.jpeg)

**Comment by diggis00** [comment]
render() {
    return (
      <View style={styles.container}>
        <Button
          title="Go to Details"
          onPress={() => {
            this.props.navigation.navigate('Home');}}   
        />
      </View>
    );
  }
}

**Comment by anastely** [comment]
Same !!

**Comment by hemadriDasari1990** [comment]
Same issue. Please find issue posted in SO

https://stackoverflow.com/questions/52784633/i-have-some-error-when-add-button-to-my-react-native-app?noredirect=1#comment92491686_52784633

**Comment by bwpAj** [comment]
Same !!  Why ???????????????????????????????????????????

**Comment by farisdewantoro** [comment]
same !! please fix it :(

**Comment by jiangpenggithub** [comment]
same!!!!!

**Comment by chornos13** [comment]
same!!!!!

**Comment by rago4** [comment]
Same here, in my case it occurs when I want to use `<Button title="String" />` component.

**Comment by chornos13** [comment]
Okay, i take a look on source code Button, and AccessibilityRole is using string, maybe this code outdated 
https://github.com/facebook/react-native/blob/bbb6a0754ce4173e24d3c0b46a5350ff2a8690d3/Libraries/Components/Button.js

maybe currently the type is needed is enum AccessibilityRole

https://github.com/facebook/react-native/blob/c5186aeb461a6eafa582b0acba1b3cebfb83550b/ReactAndroid/src/main/java/com/facebook/react/uimanager/AccessibilityDelegateUtil.java

so for you guys using component Button I suggest change to Touchable component.

**Comment by peat-psuwit** [comment]
Turns out, it happens because incomplete cherry-pick from master. Branch 0.57-stable (which I believe is where 0.57.x releases are cut from) has d3f2f96 cherry-picked (which becomes 677f1de) along with a few accessibility commits without its pre-requisites, 139559f and 1f96ff6.

https://github.com/peat-psuwit/react-native/tree/0.57-stable_android-accessibility-role_v2 contains my attempt to cherry-pick those commits. I have to revert an incorrect attempt to "Update bad method", then cherry-pick 2 aforementioned commits. I've tested this branch with my reproduction repository in the previous comment and it doesn't crash. So, I guess that should fix this bug. However, I haven't run a full test yet. And I guess I'm not supposed to send a PR to a stable branch, am I?

**Comment by manassehl9** [comment]
> Okay, i take a look on source code Button, and AccessibilityRole is using string, maybe this code outdated
> https://github.com/facebook/react-native/blob/bbb6a0754ce4173e24d3c0b46a5350ff2a8690d3/Libraries/Components/Button.js
> 
> maybe currently the type is needed is enum AccessibilityRole
> 
> https://github.com/facebook/react-native/blob/c5186aeb461a6eafa582b0acba1b3cebfb83550b/ReactAndroid/src/main/java/com/facebook/react/uimanager/AccessibilityDelegateUtil.java
> 
> so for you guys using component Button I suggest change to Touchable component.

How do I go about the changes please?

**Comment by RevoluPowered** [comment]
@peat-psuwit **"And I guess I'm not supposed to send a PR to a stable branch, am I?"**
My two cents, I know this is bad practice, but something like a hot-fix should be fine to merge with master if the basic functionality / core is broken surely? (after testing ofc) - Git rules can be broken in some cases, if others agree.

Especially if a cherry pick messed up master I'd merge the fix only with master, nothing else. 

I'm experiencing this issue too, please if you could, merge it asap as it would make me able to finish something I'm working on for work.

**Comment by RevoluPowered** [comment]
**Temporary workaround found:**
https://github.com/xotahal/react-native-material-ui/blob/master/docs/GettingStarted.md

https://github.com/xotahal/react-native-material-ui/blob/master/docs/Button.md

Just tested their buttons out and they work

**Comment by RedHotMan** [comment]
I was using a `button` component first, then I had this same issue.
I used [Touchable Opacity component](https://facebook.github.io/react-native/docs/touchableopacity) instead, and it works just fine.

**Comment by kelset** [comment]
Thanks @peat-psuwit for the detailed explanation, we'll cherry pick those two commits in the next 0.57.4 to fix this.

**Comment by peat-psuwit** [comment]
@kelset It's simply that 1592a8d is duplicating a small amount of 1f96ff6. So, I just revert that commit to avoid conflict.

**Comment by doomsower** [comment]
While 0.57.4 is not released, could someone post a patch diff here, so people like me can monkey-patch 0.57.3 with `patch-package`?

**Comment by juliandik** [comment]
Same issue when adding title to Button.

**Comment by deslee** [comment]
Is there a way for me to locally roll back to an earlier version of the package to get past the error?

**Comment by denisviklov** [comment]
here is workaround if have no time to wait when "ninjas" fix their odd work
1) delete your node_modules directory (rm -rf node_modules/)
2) npm i -S react-native@0.57.1
3) npm add @babel/runtime
4) npm i schedule@0.4.0 --save-dev
5) npm i

**Comment by robbiedh** [comment]
Same issue

**Comment by xenu256** [comment]
> here is workaround if have no time to wait when "ninjas" fix their odd work

Thanks, it works!

**Comment by rumax** [comment]
Same issue for project created with `react-native init`:

```
"dependencies": {
    "react": "16.6.0-alpha.8af6728",
    "react-native": "0.57.3"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.48.1",
    "react-test-renderer": "16.6.0-alpha.8af6728"
  },
```

**Comment by blattmann** [comment]
The workaround of @denisviklov was resolving my headache ;)
Thanks @denisviklov ! You made my day!

**Comment by ithustle** [comment]
We need the update 0.57.4 ...

**Comment by blattmann** [comment]
> The workaround of @denisviklov was resolving my headache ;)
> Thanks @denisviklov ! You made my day!

It was working for a short period of time, now I am facing the same issue again - with the previously working solution. It's super frustrating! I hope v0.57.4 is coming soon!!!

**Comment by ithustle** [comment]
Yeah ... RN is so instable ... RN dev spend more time fixing bugs than properly coding

**Comment by gluons** [comment]
Same here. Try to upgrade `react-native` to `0.57.3` but see this error. Then I back to previous version. 😰

Hope fix will be landed soon. 😦

> Sorry for notification to all subscribers 😅

**Comment by Biplovkumar** [comment]
thanks @denisviklov, really nice, its working.

**Comment by Biplovkumar** [comment]
@merbin2012 ,many errors in 0.57.3 so u need to  downgrade 0.57.1

**Comment by RevoluPowered** [comment]
I agree with the others that: The library is not stable if the default template doesn't work.

My thoughts on this are it might be easier for stability the components are in their own repository too, and maybe it's an idea to write tests for all the components, to make sure running <Button/> Renders something, then if the test fails the CI rejects the build from being merged with master. (it should prevent this happening in future too)
<View>
<etc>

It would mean showstoppers, e.g. components not working could be patched quicker, than the run-time library or vice versa (hot-fixes) and the original developers could focus on making the basics work then fix the crash bugs in the run-time.

Right now: (these figures are rough estimates)
**120 issues related to 'potential quick fix' component issues,**
**400~ relate to other stuff like the run-time errors which requires a lot of digging**

It means that the open source community could focus on fixing lots of layout/components. Which could potentially mean less work for them, because JS developers can handle the non-native device side of things, without having the hassle of building the library. _this is very useful if you have multiple run-time versions (stable, testing and beta)_

I'd like react to sort this out as I think they're overwhelmed by all the issues they are having, it's actually an okay problem to have as it means their library is getting used.

I think writing tests for the CI should be a must especially when the default template doesn't work.


Just sharing my two cents here as it helped a lot back when I worked on collaborative development work.

Anyway, _thanks to the react team for actually releasing React Native,_ and I hope they get this sorted soon.

**TL;DR:** Write tests please for all the components and don't let a release be sent to clients without a pass, make sure there is a test for every example from the documentation (implementation based and coverage based testing). Perhaps seperate the components into their own repository too, to seperate layout problems vs runtime issues

Sorry for the wall of text 👍

**Comment by esipavicius** [comment]
Always new releases with bugs... From 0.55 to 0.56 I spend 5hours on Windows and decided move to Linux. After this worked ~1hour. Now from 0.56 to 0.57.3 i spend 3hours and downgraded to 0.57.1. So wtf is this? ;D

**Comment by kelset** [comment]
Ok this issue is clearly getting out of hand so I'll lock it.

We test the releases before rolling out, and I would like to remember everyone that this is an Open Source project. The first way you can fix things is by submitting PRs to resolve your issues or fork it.

You are not paying any fee, and you are using a 0.x project. There are people spending their spare time for free in helping to make react-native better.

**Comment by gengjiawen** [comment]
0.57.5 fix this issue, please update.
